src/ln/channelmanager.rs

```diff
@@ -1487,41 +1522,42 @@ impl ChainListener for ChannelManager {
 	}
 }
 
-impl ChannelMessageHandler for ChannelManager {
-	//TODO: Handle errors and close channel (or so)
-	fn handle_open_channel(&self, their_node_id: &PublicKey, msg: &msgs::OpenChannel) -> Result<msgs::AcceptChannel, HandleError> {
-		if msg.chain_hash != self.genesis_hash {
-			return Err(HandleError{err: "Unknown genesis block hash", action: None});
-		}
-		let mut channel_state = self.channel_state.lock().unwrap();
-		if channel_state.by_id.contains_key(&msg.temporary_channel_id) {
-			return Err(HandleError{err: "temporary_channel_id collision!", action: None});
+macro_rules! handle_error {
+	($self: ident, $internal: expr, $their_node_id: expr) => {
+		match $internal {
+			Ok(msg) => Ok(msg),
+			Err(e) => {
+				match e.action {
+					Some(msgs::ErrorAction::DisconnectPeer { msg: error_msg }) => {
+						let error_msg = error_msg.unwrap();
+						$self.close_channel(&error_msg.channel_id);
+						let mut pending_events = $self.pending_events.lock().unwrap();
+						pending_events.push(events::Event::HandleError {
+							node_id: $their_node_id,
+							action: Some(msgs::ErrorAction::DisconnectPeer { msg: Some(error_msg) }),
+						});
+						return Err(false);
+					},
+					Some(msgs::ErrorAction::IgnoreError) => {
+						return Err(false);
+					},
+					Some(msgs::ErrorAction::SendErrorMessage { msg: error_msg }) => {
```

Comment:
Wait, why change the return type to Result<.., bool>? Should still be able to return the same Err object, just possibly handling the channel closure on Disconnect/SendErrorMessage first?

Reply:
Was my inquiry, do we want to give back exactly the same Err object even if action is no more accurate? At first I've written return { HandleError: ...action : msgs::ErrorAction::IgnoreError } for DisconnectPeer and SendErrorMessage. 
In the wrapper we just manage channel closure, not the others errors actions given back?

Reply:
Yea, I assume just return the same Err, since the ErrorMessage still needs to be relayed to the peer.